[ADD] add a new module to allow one customer to be shared among

duanyp1991 opened the pull request:
```
some of the sibling companies.
```

 Changes to be committed:

```
new file:   multi_company_partner/README.rst
new file:   multi_company_partner/__init__.py
new file:   multi_company_partner/__openerp__.py
new file:   multi_company_partner/partner.py
new file:   multi_company_partner/partner_view.xml
new file:   multi_company_partner/security/ir.model.access.csv
new file:   multi_company_partner/security/multi_company_orders.xml
new file:   multi_company_partner/test/test_multi_company_partner.yml
```

Commit:
[ADD] add a new module to allow one customer to be shared among
    some of the sibling companies.
 Changes to be committed:
	new file:   multi_company_partner/README.rst
	new file:   multi_company_partner/__init__.py
	new file:   multi_company_partner/__openerp__.py
	new file:   multi_company_partner/partner.py
	new file:   multi_company_partner/partner_view.xml
	new file:   multi_company_partner/security/ir.model.access.csv
	new file:   multi_company_partner/security/multi_company_orders.xml
	new file:   multi_company_partner/test/test_multi_company_partner.yml

pedrobaeza commented:
Good addition!

Please name it the other way: _partner_multi_company_.

pedrobaeza commented on multi_company_partner/README.rst:
```diff
@@ -0,0 +1,74 @@
+    some of the sibling companies.
+
+Standard Odoo limitation:
+    one customer can be only private to one company or shared among
```
s/can be only/can only be

pedrobaeza commented on multi_company_partner/README.rst:
```diff
@@ -0,0 +1,74 @@
+=====================
+
+This basic module aims to allow one customer to be shared among
+    some of the sibling companies.
```
s/to be shared among.../to be shared among several companies, not all or only one.

pedrobaeza replied:
There's no need to indent second and successive paragraph lines

duanyp1991 replied:
That's quite efficient! @pedrobaeza  Thanks!

pedrobaeza commented on multi_company_partner/README.rst:
```diff
@@ -0,0 +1,74 @@
+Installation
+============
+
+To install this module, you need to:
```
You can put: `Install the module the regular way.`

pedrobaeza commented on multi_company_partner/README.rst:
```diff
@@ -0,0 +1,74 @@
+Configuration
+=============
+
+To configure this module, you need to:
```
Put `Nothing special to configure.`

pedrobaeza commented on multi_company_partner/README.rst:
```diff
@@ -0,0 +1,74 @@
+=====
+ A new field is introduced on partner form: Allowed companies under field 'System company'.
+ everytime the salesperson creates a new customer,
+ he can add allowed companies for this customer.The salesperson has to belong
```
s/, he/. You/

pedrobaeza commented on multi_company_partner/__openerp__.py:
```diff
@@ -0,0 +1,113 @@
+ 'author': 'Elico Corp, Odoo Community Association (OCA)',
+ 'license': 'AGPL-3',
+ 'website': 'https://www.elico-corp.com',
+ 'description': """
```
Remove this description. It's already in README.rst

pedrobaeza commented on multi_company_partner/partner.py:
```diff
@@ -0,0 +1,53 @@
+class ResPartner(orm.Model):
+        _inherit = 'res.partner'
+
+        _columns = {
```
You can use new API for this little piece of code.

duanyp1991 replied:
Hi, this is a pull request for v7, is there a new API for v7 to define a new field for a model? Thanks.

pedrobaeza replied:
Sorry, I though it was for v8. I have said nothing then (and don't remove description key as well).

duanyp1991 replied:
I'll make another pull request for v8 anyway;)

pedrobaeza commented on multi_company_partner/partner.py:
```diff
@@ -0,0 +1,53 @@
+            '''Initializes the company_ids with the existing company_id.
+
+            '''
+            cr.execute('insert into res_company_partner_rel '
```
This is now better done on post_migration hook. You can see an example here:

https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control_stock/__openerp__.py#L42
https://github.com/odoomrp/odoomrp-wip/blob/8.0/quality_control_stock/__init__.py#L10

duanyp1991 replied:
Thanks for the tip! Does the post_migration work for v7?

pedrobaeza replied:
This is also only for v8. You can put anyway this as a migration script, because this is executed each time the server is launched.

duanyp1991 replied:
OK, I'll remove the method.:)

pedrobaeza replied:
Do you know how to do the migration script?

duanyp1991 replied:
For v7 , no. Is it related to OpenUpgrade?

pedrobaeza replied:
No, it's something standard. You can execute a migration script the first time you install the module. Each migration script has two params: cr and version. The second one is put to None when installing the module, so you can take advantage of this to make this assignation the first time.

duanyp1991 replied:
Thanks for your answer! However I'm still not sure how I can use the migration script, is there any doc or example I can find somewhere?

Is there any folder structure/ naming convention for migration scripts for community module?

pedrobaeza replied:
Here you can see an example of a migration script with `if not version:` condition that fits for first install:

https://github.com/OCA/account-financial-reporting/blob/7.0/account_financial_report_webkit/migrations/7.0.1.0.2/post-migration.py

pedrobaeza commented on multi_company_partner/partner.py:
```diff
@@ -0,0 +1,53 @@
+            '''
+            cr.execute('insert into res_company_partner_rel '
+                       '(select id, company_id from res_partner '
+                       'where id not in (select distinct partner_id '
```
Exclude partner_id = False

pedrobaeza replied:
Sorry, company_id = False

Commit:
[IMP] - fix description
      - change module name

Commit:
[ADD] add migration script.
 Changes to be committed:
	modified:   partner_multi_company/__openerp__.py
	new file:   partner_multi_company/migrations/7.0.0.0/post-migration.py

lasley commented:
This can be closed I think - not sure the history, but `partner_multi_company` is already in future branches & 7 is now deprecated
